### ISSUE TITLE
Adding Callbacks

### ISSUE THREAD

**Issue by Optimox** [dreamquark-ai/tabnet#123]
# Main Problem

Currently some things can be changed like scheduler or optimizer but it's not possible to do things like changing the loss function, the early stopping metrics and probably some important things for specific problems.

# Proposed Solutions

We should find a simple way of using callbacks in order to customize more the training process.
Something that would resemble one of these:
- https://github.com/ncullen93/torchsample
- https://pytorch-lightning.readthedocs.io/en/latest/callbacks.html

The easier it is and the less invasive solution for the code the better

If you feel like this is interesting and would like to contribute, please share your ideas in comments or open a PR!

**Comment by Optimox** [comment]
Done by @queraq in #193 thanks!
